course/include/isometry/vector3.hpp

```diff
@@ -0,0 +1,152 @@
+  static const Vector3 kUnitX;
+  static const Vector3 kUnitY;
+  static const Vector3 kUnitZ;
+
+  /// \brief Const implementation of the sum operator.
+  Vector3 operator+(const Vector3& vector) const;
+
+  /// \brief Const implementation of the sub operator.
+  Vector3 operator-(const Vector3& vector) const;
+
+  /// \brief Const implementation of the mult operator.
+  Vector3 operator*(const Vector3& vector) const;
+
+  /// \brief Const implementation of the mult times double operator.
+  Vector3 operator*(double scalar) const;
+
+  /// \brief Const implementation of the mult times integer operator.
+  Vector3 operator*(int scalar) const;
+
+  /// \brief Const implementation of the over integer operator.
+  Vector3 operator/(const Vector3& vector) const;
+
+  /// \brief Const implementation of the mult times double operator.
+  Vector3 operator/(double scalar) const;
+
+  /// \brief Non const implementation of the plus assign operator.
+  Vector3& operator+=(const Vector3& vector);
+
+  /// \brief Non const implementation of the minus assign operator.
+  Vector3& operator-=(const Vector3& vector);
+
+  /// \brief Non const implementation of the mult times vector assign operator.
+  Vector3& operator*=(const Vector3& vector);
+
+  /// \brief Non const implementation of the mult times double assign operator.
+  Vector3& operator*=(double scalar);
+
+  /// \brief Non const implementation of the divide over vector assign operator.
+  Vector3& operator/=(const Vector3& vector);
+
+  /// \brief Non const implementation of the divide over double assign operator.
+  Vector3& operator/=(double scalar);
+
+  /// \brief Equals to operator.
+  bool operator==(const Vector3& vector) const;
+
+  /// \brief Non-equals to operator.
+  bool operator!=(const Vector3& vector) const;
+
+  /// \brief Const implementation of the [] accessor.
+  /// \return An rval copy of the requested field.
+  double operator[](int index) const;
+
+  /// \brief Non-const implementation of the [] accessor.
+  /// \return A mutable reference to the requested field.
+  double& operator[](int index);
+
+  /// \brief Dot product between this and a given vector.
+  double dot(const Vector3& vector) const;
+
+  /// \brief Cross product between this and a given vector.
+  Vector3 cross(const Vector3& vector) const;
+
+  /// \brief Calculates the norm of this vector.
+  /// \return The norm of the vector.
+  double norm() const;
+
+  /// \brief Getter of x.
+  /// \return An rval copy of x.
+  double x() const;
+
+  /// \brief Getter of y.
+  /// \return An rval copy of y.
+  double y() const;
+
+  /// \brief Getter of z.
+  /// \return An rval copy of z.
+  double z() const;
+
+  /// \brief Getter of x.
+  /// \return A mutable reference to x.
+  double& x();
+
+  /// \brief Getter of y.
+  /// \return A mutable reference to y.
+  double& y();
+
+  /// \brief Getter of z.
+  /// \return A mutable reference to z.
+  double& z();
+
+  /// \brief Setter of x.
+  void x(double _x);
```

Comment:
Remove these setters. The ones above are bidirectional when not `const`, so there's no need for explicit setters. Also, if they were necessary (not in this case) explicit setter methods should clearly state the side-effect on the name, such as in `void setX(double x)`.